servicenow_connector.py

```diff
@@ -1653,6 +1664,83 @@ def _query_users(self, param):
 
         return result
 
+    def _search_sources_details(self, action_result, sysparm_term, sysparm_search_sources):
+
+        ret_val, auth, headers = self._get_authorization_credentials(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.set_status(phantom.APP_ERROR, SERVICENOW_AUTH_ERROR_MESSAGE)
+
+        params = {"sysparm_term": sysparm_term, "sysparm_search_sources": sysparm_search_sources}
+        params['sysparm_page'] = SERVICENOW_DEFAULT_PAGE
+        params['sysparm_limit'] = SERVICENOW_MAX_LIMIT
+
+        items_list = []
+        result_length = 0
+        first_call = True
+        total_result_count_page_limit = 0
+
+        while True:
+            ret_val, response = self._make_rest_call_helper(
+                action_result, SERVICENOW_SEARCH_SOURCE_ENDPOINT, auth=auth, headers=headers, params=params
+            )
+            if phantom.is_fail(ret_val):
+                self.debug_print(action_result.get_message())
+                return action_result.set_status(phantom.APP_ERROR, action_result.get_message())
+
+            total_item_count = int(response.get("result", {}).get("result_count", 0))
+            search_results_len = len(response.get("result").get("search_results", []))
+            for i in range(search_results_len):
+                response.get("result").get("search_results", [])[i].pop("limit")
+                response.get("result").get("search_results", [])[i].pop("page")
+                result_length += len(response.get("result").get("search_results", [])[i].get("records", []))
+
+            # In first call I want to get response['result'] and in other calls I am extending records into result
+            # In first call we are adding total pages that need to iterate. Because, we are getting empty record in result due to ACLs
+            # Getting default/maximum records from servicenow is 20 in one page
```

Comment:
can we perhaps make this comment more succinct?

Reply:
sure, I have succinct the comment.